From 6dd2259acd4424995ba38b8f44b2a2adb82102af Mon Sep 17 00:00:00 2001
From: Cyril Richard <cyril.richard42@gmail.com>
Date: Sun, 13 Jul 2025 23:52:33 +0200
Subject: [PATCH 1/2] Fix matrix element access in cvTransformImageRefPoint

Replace 2D matrix access at<double>(0,1) with 1D vector access
at<double>(1) since H * Mat(refptin) produces a 3x1 column vector.
The incorrect 2D access was causing OpenCV assertion failures.
---
 src/opencv/opencv.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/opencv/opencv.cpp b/src/opencv/opencv.cpp
index dfbe361db6..c7b2c87505 100644
--- a/src/opencv/opencv.cpp
+++ b/src/opencv/opencv.cpp
@@ -322,8 +322,8 @@ void cvTransformImageRefPoint(Homography Hom, point refpointin, point *refpointo
 	Mat H = Mat(3, 3, CV_64FC1);
 	convert_H_to_MatH(&Hom, H);
 	refptout = H * Mat(refptin);
-	refpointout->x = refptout.at<double>(0, 0);
-	refpointout->y = refptout.at<double>(0, 1);
+	refpointout->x = refptout.at<double>(0);
+	refpointout->y = refptout.at<double>(1);
 }
 
 static void convert_H_to_MatH(const Homography *from, Mat &to) {
-- 
GitLab
